rules/prefer-event-target.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const MESSAGE_ID = 'prefer-event-target';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `EventTarget` over `EventEmitter`.',
+};
+
+const eventEmitterClassSelector = [
+	':matches(ClassDeclaration, ClassExpression)',
+	'[superClass.name="EventEmitter"]',
+	'[body.type="ClassBody"]',
+].join('');
+
+const newEventEmitterSelector = [
+	'NewExpression',
+	'[callee.name="EventEmitter"]',
+].join('');
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = () => ({
+	[eventEmitterClassSelector](node) {
+		return {
+			node,
+			messageId: MESSAGE_ID,
+		};
+	},
+	[newEventEmitterSelector](node) {
```

Comment:
If these two listeners have the same logic, you can `matches()` from `selectors/index.js` to combine these two selectors into one.